traitsui/testing/tests/test_exception_handling.py

```diff
@@ -0,0 +1,74 @@
+#  Copyright (c) 2008-20, Enthought, Inc.
+#  All rights reserved.
+#
+#  This software is provided without warranty under the terms of the BSD
+#  license included in LICENSE.txt and may be redistributed only
+#  under the conditions described in the aforementioned license.  The license
+#  is also available online at http://www.enthought.com/licenses/BSD.txt
+#
+#  Thanks for using Enthought open source!
+#
+
+""" Tests for traitsui.testing.exception_handling """
+
+import unittest
+
+from pyface.api import GUI
+
+from traits.api import HasTraits, Int
+from traitsui.tests._tools import (
+    requires_toolkit,
+    ToolkitName,
```

Comment:
will these eventually be moved into `traitsui.testing`?  I feel like they don't need to if they aren't a dependency outside of the tests in `traitsui.testing`, correct?

Reply:
> will these eventually be moved into traitsui.testing?

I am not planning to.

I think I want to think of `traitsui.testing` being used by both traitsui AND projects using of traitsui.
requires_toolkit and ToolkitName are utilities only used by traitsui's own tests, so by keeping them out of the way of `traitsui.testing`, it is more obvious to developers that we can change them without worrying about affecting downstream projects in any way.

That is why I did not reuse `is_qt` in `traitsui.testing.gui`. It is such a teeny tiny function that does not warrant the additional coupling.

>  I feel like they don't need to if they aren't a dependency outside of the tests in traitsui.testing, correct?

I am not sure if I understand the question fully. Perhaps part of this was my choice of wording in #1238
In #1238, I wrote:
> Currently a few functions in traitsui.tests._tools are imported by traitsui.testing:

When I wrote "imported by `traitsui.testing`", I have in my mind excluded `traitsui.testing.*.tests.*`.
My concern in #1238 was mainly about having externally facing testing library behaviour being implemented in a private module that is obviously internally facing. `traitsui.testing.*.tests.*` are not library, so the concern does not apply there.